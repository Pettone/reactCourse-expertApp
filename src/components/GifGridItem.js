import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( { id, title, url} ) => {

    console.log({id,title,url});

    return (
        <div className = "card animate__animated animate__bounce animate__delay-2s">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
