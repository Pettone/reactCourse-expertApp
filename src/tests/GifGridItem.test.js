import React from 'react';
import { render } from '@testing-library/jest-dom';
import { GifGridItem } from '../components/GifGridItem'; 
import { shallow } from 'enzyme';


describe('Testing GifGridItem component', () => {
    
    const id = '1';
    const title = 'Hi, I am a title';
    const url = 'https://www.url.com/image';
    const wrapper = shallow( <GifGridItem id = {id} title = {title} url = {url}/> );

    test('Must to show component correctly', () => { 

        expect(wrapper).toMatchSnapshot();
    });

    test('Must to have a paragraph', () => {

        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(title);
    });

    test('Must to have an image and url equal to props', () => {

        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe(url);
        expect( img.prop('alt') ).toBe(title);
    });

    test('Must to have animate__bounce animate__delay-2s class...', () => {

        const div = wrapper.find('div');
        const className = div.prop('className');
        expect( className.includes('animate__bounce animate__delay-2s')).toBe(true);
    });
});


