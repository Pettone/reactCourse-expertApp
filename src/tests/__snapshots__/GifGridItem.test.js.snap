// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem component Must to show component correctly 1`] = `
<div
  className="card animate__animated animate__bounce animate__delay-2s"
>
  <img
    alt="Hi, I am a title"
    src="https://www.url.com/image"
  />
  <p>
    Hi, I am a title
  </p>
</div>
`;
